banzai_floyds/extract.py

```diff
@@ -1,183 +1,34 @@
-        profile_width = profile_widths[order_id - 1](wavelength_bin)
-        peak = np.argmin(np.abs(profile_center - data_to_fit['y_order']))
-
-        # Pass a match filter (with correct s/n scaling) with a gaussian with a default width
-        initial_coeffs = np.zeros((x_poly_order + 1) + y_poly_order)
-        initial_coeffs[0] = np.median(data_to_fit['data']) / data_to_fit['data'][peak]
-        # TODO: Fit the background with a totally fixed profile, and no need to iterate
-        # since our filter is linear
-        background_window = np.abs(data_to_fit['y_order'] - profile_center) > 3.0 * fwhm_to_sigma(profile_width)
-        model = models.Legendre2D(x_degree=x_poly_order, y_degree=y_poly_order,
-                                  x_domain=(np.min(data_to_fit['wavelength'][background_window]),
-                                            np.max(data_to_fit['wavelength'][background_window])),
-                                  y_domain=(np.min(data_to_fit['y_order'][background_window]),
-                                            np.max(data_to_fit['y_order'][background_window])))
-        inv_variance = data_to_fit['uncertainty'][background_window] ** -2.0
-        best_fit_model = fitter(model,
-                                x=data_to_fit['wavelength'][background_window],
-                                y=data_to_fit['y_order'][background_window],
-                                z=data_to_fit['data'][background_window],
-                                weights=inv_variance)
-        background = best_fit_model(data_to_fit['wavelength'], data_to_fit['y_order'])
-        background_fit = Table({'x': data_to_fit['x'],
-                                'y': data_to_fit['y'],
-                                'background': background})
-        results = vstack([background_fit, results])
-    return results
-
-
-def get_wavelength_bins(wavelengths):
-    """
-    Set the wavelength bins to be at the pixel edges along the center of the orders.
-    """
-    # TODO: in the long run we probably shouldn't bin at all and just do a full 2d sky fit
-    #   (including all flux in the order, yikes)
-    # Throw out the edge bins of the order as the lines are tilt and our orders are vertical
-    pixels_to_cut = np.round(0.5 * np.sin(np.deg2rad(wavelengths.line_tilts)) * wavelengths.orders.order_heights)
-    pixels_to_cut = pixels_to_cut.astype(int)
-    bin_edges = wavelengths.bin_edges
-    cuts = []
-    for cut in pixels_to_cut:
-        if cut == 0:
-            right_side_slice = slice(1, None)
-        else:
-            right_side_slice = slice(1+cut, -cut)
-        left_side_slice = slice(cut, -1-cut)
-        cuts.append((right_side_slice, left_side_slice))
-    return [Table({'center': (edges[right_cut] + edges[left_cut]) / 2.0,
-                   'width': edges[right_cut] - edges[left_cut]})
-            for edges, (right_cut, left_cut) in zip(bin_edges, cuts)]
+def set_extraction_region(image):
```

Comment:
I added a helper function to turn the extraction windows into pixels that I include in the extraction region to make that logic more testable.